pkg/resources/digitalocean/cloud.go

```diff
@@ -175,3 +183,97 @@ func (c *Cloud) GetApiIngressStatus(cluster *kops.Cluster) ([]kops.ApiIngressSta
 
 	return nil, nil
 }
+
+// FindClusterStatus discovers the status of the cluster, by looking for the tagged etcd volumes
+func (c *Cloud) FindClusterStatus(cluster *kops.Cluster) (*kops.ClusterStatus, error) {
+	etcdStatus, err := findEtcdStatus(c, cluster)
+	if err != nil {
+		return nil, err
+	}
+	status := &kops.ClusterStatus{
+		EtcdClusters: etcdStatus,
+	}
+	klog.V(2).Infof("Cluster status (from cloud): %v", fi.DebugAsJsonString(status))
+	return status, nil
+}
+
+// findEtcdStatus discovers the status of etcd, by looking for the tagged etcd volumes
+func findEtcdStatus(c *Cloud, cluster *kops.Cluster) ([]kops.EtcdClusterStatus, error) {
+	statusMap := make(map[string]*kops.EtcdClusterStatus)
+	volumes, err := getAllVolumesByRegion(c, c.RegionName)
+
+	if err != nil {
+		return nil, fmt.Errorf("error describing volumes: %v", err)
+	}
+
+	for _, volume := range volumes {
+		volumeID := volume.ID
+
+		etcdClusterName := ""
+		var etcdClusterSpec *etcd.EtcdClusterSpec
+
+		for _, myTag := range volume.Tags {
+			klog.V(2).Infof("findEtcdStatus status (from cloud): %v", myTag)
+			// check if volume belongs to this cluster.
+			// tag will be in the format "KubernetesCluster:dev5-k8s-local" (where clusterName is dev5.k8s.local)
+			clusterName := strings.Replace(cluster.Name, ".", "-", -1)
+			if strings.Contains(myTag, fmt.Sprintf("%s, %s", TagKubernetesClusterNamePrefix, ":", clusterName)) {
```

Comment:
Am I misreading this one, or does the call to `fmt.Sprintf` use 3 variables but only 2 placeholders?

Reply:
That was a miss - I updated it as part of the future commit.